include/forcing/GridDataSelector.hpp

```diff
@@ -0,0 +1,300 @@
+    //     const auto ydiff = static_cast<double>(grid.rows) / (grid.extent.ymax - grid.extent.ymin);
+    //     const auto xdiff = static_cast<double>(grid.columns) / (grid.extent.xmax - grid.extent.xmin);
+    //     const auto bbox = bounding_box_(polygon);
+    // 
+    //     for (const auto& p : polygon.outer()) {
+    //         const auto x_index = position_(p.get<0>(), grid.extent.xmin, grid.extent.xmax, grid.columns);
+    //         const auto y_index = position_(p.get<1>(), grid.extent.ymin, grid.extent.ymax, grid.rows);
+    //         const auto set_pair = boundary.emplace(x_index, y_index, 0, NAN);
+    //     }
+    // 
+    //     throw std::runtime_error{"Boundary-constructor not implemented"};
+    // }
+
+    /**
+     * Extent-based constructor
+     *
+     * @param config Selector configuration options
+     * @param grid Source grid specification
+     * @param extent Target bounding box used as mask
+     */
+    GridDataSelector(
+      SelectorConfig config,
+      const GridSpecification& grid,
+      const Extent& extent
+    ) noexcept
+      : config_(std::move(config))
+    {
+        const auto col_min = position_(extent.xmin, grid.extent.xmin, grid.extent.xmax, grid.columns);
+        const auto col_max = position_(extent.xmax, grid.extent.xmin, grid.extent.xmax, grid.columns);
+        const auto row_min = position_(extent.ymin, grid.extent.ymin, grid.extent.ymax, grid.rows);
+        const auto row_max = position_(extent.ymax, grid.extent.ymin, grid.extent.ymax, grid.rows);
+        const auto ncells  = (row_max - row_min) * (col_max - col_min);
+
+        cells_.reserve(ncells);
+        for (auto row = row_min; row < row_max; row++) {
+            for (auto col = col_min; col < col_max; col++) {
+                cells_.emplace_back(Cell{/*x=*/col, /*y=*/row, /*z=*/0UL, /*value=*/NAN});
+            }
+        }
+    }
+
+    GridDataSelector() noexcept = default;
+
+    virtual ~GridDataSelector() = default;
+
+    time_t& initial_time() noexcept {
+        return config_.init_time;
+    }
+
+    time_t initial_time() const noexcept {
+        return config_.init_time;
+    }
+
+    long& duration() noexcept {
+        return config_.duration_seconds;
+    }
+
+    long duration() const noexcept {
+        return config_.duration_seconds;
+    }
+
+    std::string& variable() noexcept {
+        return config_.variable_name;
+    }
+
+    const std::string& variable() const noexcept {
+        return config_.variable_name;
+    }
+
+    std::string& units() noexcept {
+        return config_.variable_units;
+    }
+
+    const std::string& units() const noexcept {
+        return config_.variable_units;
+    }
+
+    boost::span<Cell> cells() noexcept {
+        return cells_;
+    }
+
+    boost::span<const Cell> cells() const noexcept {
+        return cells_;
+    }
+
+  private:
+    //! Returns true if point is inside polygon or on its boundary.
+    //! @note may not be needed
```

Comment:
It doesn't look like this gets called, currently. Maybe it was meant for use in the boundary-based constructor. Couldn't something from Boost Geometry be used instead of implementing this ourselves?